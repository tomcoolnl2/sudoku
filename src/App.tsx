
import { FC, lazy, Suspense, useContext } from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { PersistGate } from 'redux-persist/integration/react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { ErrorBoundary } from 'react-error-boundary'
import { configureStore } from './redux'	
import { DarkThemeProvider } from './styles/core/ThemeProvider'
import * as Styled from './styles'
import { Menu, ErrorFallback } from './components'
import { PeopleContext, ActionType, People } from './context/people'

const Splash = lazy(() => import('./pages/splash'))
const Game = lazy(() => import('./pages/game'))
const Settings = lazy(() => import('./pages/settings'))

// Weird validation error causes the app to crash, but it works fine
// eslint-disable-next-line
// @ts-ignore
const { store, persistor } = configureStore()

export const App: FC = () => {

	const { state, dispatch } = useContext(PeopleContext)
	
	console.log(state, dispatch)
	
	function devour(name: string): void {
		dispatch({ type: ActionType.CHOMP, payload: name })
	}

	function spitOut(name: string): void {
		dispatch({ type: ActionType.REVIVE, payload: name })
	}

	return (	
		<ReduxProvider store={store}>
			<PersistGate loading={null} persistor={persistor}>
				<DarkThemeProvider>
					<Router>
						<Styled.Content>
							<Menu />
							<Styled.Title>Sudoku</Styled.Title>
							<Styled.Card>
								<People>
									HELLO {}
									{state.map((person, idx) => (
										<div key={idx} style={{ display: 'flex', width: '50%', justifyContent: 'space-around' }}>
											<div>{person.name}</div>
											{person.alive ?
												<div> ‚ú®‚ú® ALIVE! ‚ú®‚ú® <button onClick={() => devour(person.name)}> üêä DEVOUR üêä </button> </div> :
												<div> ‚ò† ‚ò† DEAD ‚ò† ‚ò† <button onClick={() => spitOut(person.name)}> ü•µ SPIT OUT ü•µ </button> </div>}
										</div>
									))}
								</People>
								<ErrorBoundary FallbackComponent={ErrorFallback} onReset={location.reload}>
									<Suspense fallback={<div>Loading...</div>}>
										<Switch>
											<Route exact path="/">
												<Splash />
											</Route>
											<Route path="/game">
												<Game />
											</Route>
											<Route path="/settings">
												<Settings />
											</Route>
										</Switch>
									</Suspense>
								</ErrorBoundary>
							</Styled.Card>
						</Styled.Content>
					</Router>
				</DarkThemeProvider>
			</PersistGate>
		</ReduxProvider>
	)
}