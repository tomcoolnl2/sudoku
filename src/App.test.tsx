
import { render } from '@testing-library/react'
import { App } from './App'


describe('App', () => {
	
	it('renders with light mode as default', () => {
		const { container } = render(<App />)
		expect(container.firstChild).toMatchSnapshot()
	})
})