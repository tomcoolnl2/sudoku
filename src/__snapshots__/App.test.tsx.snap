// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders with light mode as default 1`] = `
.c1 {
  color: 'white';
  margin-top: 0;
  text-align: center;
}

.c0 {
  max-width: 500px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  background-color: white;
  border-radius: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: -webkit-fit-content;
  max-height: -moz-fit-content;
  max-height: fit-content;
  padding: 15px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #282c34;
  border: 2px solid #282c34;
  border-radius: 4px;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 4px 2px;
  min-height: 40px;
  padding: 0;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  opacity: .9;
}

.c3:disabled {
  opacity: .7;
  cursor: default;
}

.c3:focus {
  border-color: #EDDA74;
  outline: none;
}

.c3:hover {
  opacity: .7;
}

.c3 > svg {
  margin-right: 10px;
  fill: white;
  position: relative;
  width: 24px;
  height: 24px;
}

<div
  class="c0"
>
  <ul>
    <li>
      <a
        href="/"
      >
        Start
      </a>
    </li>
    <li>
      <a
        href="/settings"
      >
        Settings
      </a>
    </li>
  </ul>
  <h1
    class="c1"
  >
    Sudoku
  </h1>
  <div
    class="c2"
  >
    <label
      for="mistakes-limit"
    >
      Allowed Mistakes:
    </label>
    <select
      id="mistakes-limit"
      name="mistakes-limit"
    >
      <option
        value="2"
      >
        Hard: 2
      </option>
      <option
        value="3"
      >
        Medium: 3
      </option>
      <option
        selected=""
        value="5"
      >
        Easy: 5
      </option>
    </select>
    <a
      href="/game"
    >
      <button
        class="c3"
      >
        Todo!
      </button>
    </a>
  </div>
</div>
`;
